agents/holo_synth/synth_agent.py

```diff
@@ -0,0 +1,208 @@
+import argparse
+import os
+import time
+
+import numpy as np
+import txaio
+import yaml
+
+ON_RTD = os.environ.get("READTHEDOCS") == "True"
+if not ON_RTD:
+    from holog_daq import synth3
+    from ocs import ocs_agent, site_config
+    from ocs.ocs_twisted import Pacemaker, TimeoutLock
+
+
+class SynthAgent:
+    """
+    Agent for connecting to the Synths for holography
+    """
+
+    def __init__(self, agent, config_file):
+
+        self.lo_id = None
+        self.initialized = False
+        self.take_data = False
+
+        self.agent = agent
+        self.log = agent.log
+        self.lock = TimeoutLock()
+
+        agg_params = {"frame_length": 10 * 60}  # [sec]
+        self.agent.register_feed(
+            "synth_LO", record=True, agg_params=agg_params, buffer_time=0
+        )
+        """
+        if mode == 'acq':
+            self.auto_acq = True
+        else:
+            self.auto_acq = False
+        self.sampling_frequency = float(samp)
+
+        ### register the position feeds
+        """
+        if config_file == "None":
+            raise Exception("No config file specified for the FTS mirror config")
+        else:
+            config_file_path = os.path.join(os.environ["OCS_CONFIG_DIR"], config_file)
+            print(config_file_path)
+            with open(config_file_path) as stream:
+                self.holog_configs = yaml.safe_load(stream)
+                if self.holog_configs is None:
+                    raise Exception("No mirror configs in config file.")
+                self.log.info(f"Loaded mirror configs from file {config_file_path}")
+                self.N_MULT = self.holog_configs.pop("N_MULT", None)
+                self.ghz_to_mhz = self.holog_configs.pop("ghz_to_mhz", None)
+
+    def init_synth(self, session, params=None):
+        """init_synth(params=None)
+        Perform first time setup for communication with Synth.
+        """
```

Comment:
This formatting is close. Since this doesn't have any params we want that first line to read `init_synth()`. This method signature is what client writers will read in the documentation, and passing a literal "params" argument to the client is not what we want here.

Should be identified as a "Task" in the docstring. Example here: https://ocs.readthedocs.io/en/develop/developer/agents.html#task-and-process-documentation

Reply:
Needs some more small adjustments. Currently you have:
```
        """init_synth(params=None)

        **Task** - A task to initialize the synthesizers.

        Parameters: None

        Examples:
            Example for calling in a client::
                import ocs
                from ocs.ocs_client import OCSClient
                agent = OCSClient("synth_LO")
                agent.init_synth()

        Notes:
            This task is called to turn on the synthesizers.
        """
```

Changes to make:
- First line should be changed to `init_synth()`
- `Parameters: None` line should be removed
- Blank line needed after the double `::` in the Examples section (this'll format the code block properly)
- `import ocs` can be removed from the example code